packages/botonic-nlp/src/tasks/text-classification/process/processor.ts

```diff
@@ -0,0 +1,61 @@
+import { tensor } from '@tensorflow/tfjs-node'
+
+import { Sample } from '../../../dataset/types'
+import { LabelEncoder } from '../../../encode/label-encoder'
+import { OneHotEncoder } from '../../../encode/one-hot-encoder'
+import { PADDING_TOKEN, UNKNOWN_TOKEN } from '../../../preprocess/constants'
+import { Preprocessor } from '../../../preprocess/preprocessor'
+import { InputData, OutputData } from './types'
+
+export class Processor {
+  constructor(
+    readonly preprocessor: Preprocessor,
+    readonly tokensEncoder: LabelEncoder,
+    readonly classEncoder: OneHotEncoder
+  ) {}
+
+  process(samples: Sample[]): { x: InputData; y: OutputData } {
+    const processedSamples = samples.map(s => this.processSample(s))
+    return {
+      x: tensor(processedSamples.map(s => s.x)),
+      y: tensor(processedSamples.map(s => s.y)),
+    }
+  }
+
+  private processSample(sample: Sample): { x: number[]; y: number[] } {
+    return {
+      x: this.generateSampleInput(sample.text),
+      y: this.generateSampleOutput(sample.class),
+    }
+  }
+
+  private generateSampleInput(text: string): number[] {
+    const sequence = this.generateSequence(text)
+    return this.tokensEncoder.encode(sequence)
+  }
+
+  private generateSequence(text: string): string[] {
+    const normalizedText = this.preprocessor.normalize(text)
```

Comment:
DRY: Duplicated in VocabularyGenerator.generate i a TextProcessor.generateSequence. Create Preprocessor.full(string): string[]?

Reply:
You are right, maybe something like `preprocessor.preprocess()` would be nice to have. I've opened a [ticket](https://linear.app/hubtype/issue/BOT-327/preprocessor-generic-method) for this.